libs/platform/src/lib/components/table/table.component.ts

```diff
@@ -80,21 +103,56 @@ interface SelectableRow {
 @Component({
     selector: 'fdp-table',
     templateUrl: './table.component.html',
-    changeDetection: ChangeDetectionStrategy.OnPush
+    styleUrls: ['./table.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [TableService],
+    host: {
+        '[class.fd-table]': 'true',
+        '[class.fd-table--compact]': 'contentDensity === CONTENT_DENSITY.COMPACT',
+        '[class.fd-table--condensed]': 'contentDensity === CONTENT_DENSITY.CONDENSED',
+        '[class.fd-table--no-horizontal-borders]': 'noHorizontalBorders || noBorders',
+        '[class.fd-table--no-vertical-borders]': 'noVerticalBorders || noBorders'
+    }
 })
-export class TableComponent implements OnChanges, OnInit, AfterViewInit, OnDestroy, CssClassBuilder {
+export class TableComponent<T = any> implements AfterViewInit, OnDestroy, OnChanges {
     /** Data source for table data. */
     @Input()
-    set dataSource(value: FdpTableDataSource<any>) {
+    set dataSource(value: FdpTableDataSource<T>) {
         if (value) {
-            this._reset();
+            this._resetChecks();
             this._initializeDS(value);
         }
     }
-    get dataSource(): FdpTableDataSource<any> {
-        return this._dataSource;
+    get dataSource(): FdpTableDataSource<T> {
+        return this._ds;
     }
 
+    /** Initial state of table. */
+    @Input()
+    set state(value: TableState) {
+        if (!value) {
+            this.setTableState(DEFAULT_TABLE_STATE);
+        } else {
+            this.setTableState(value);
+        }
```

Comment:
could be simplified
`this.setTableState(value || DEFAULT_TABLE_STATE)`

Reply:
fixed